src/files.ts

```diff
@@ -141,14 +153,37 @@ async function* searchFiles(
   symlinksEnabled: boolean,
   ignores: string[],
 ): AsyncGenerator<string | Buffer> {
+  const positiveIgnores = ignores.filter(rule => !rule.startsWith('!'));
+  const negativeIgnores = ignores.filter(rule => rule.startsWith('!')).map(rule => rule.substring(1));
   const searcher = fg.stream(patterns, {
     ...fgOptions,
     cwd,
     followSymbolicLinks: symlinksEnabled,
+    ignore: positiveIgnores,
   });
   for await (const filePath of searcher) {
-    if (filterIgnoredFiles([filePath.toString()], ignores).length) {
-      yield filePath;
+    yield filePath;
+  }
+  // TODO: This is incorrect because the .gitignore format allows to specify exceptions to previous rules, therefore
+  // the separation between positive and negative ignores is incorrect in a scenario with 2+ exeptions like the one below:
+  // `node_module/` <= ignores everything in a `node_module` folder and it's relative subfolders
+  // `!node_module/my_module/` <= excludes the `my_module` subfolder from the ignore
+  // `node_module/my_module/build/` <= re-includes the `build` subfolder in the ignore
+  if (negativeIgnores.length) {
+    const searcher2 = fg.stream(negativeIgnores, {
```

Comment:
required: `searcher2` :D I suggest splitting this into separate function to make it a little bit more readable.